Carousel: CSS tweaks to ensure WPCOM compatibility

ramonjd opened the pull request:
#### Changes proposed in this Pull Request

While testing D63016-code, we noticed a couple of minor UI irregularities:

**1. Font color is missing on the default post comment button**
<img width="905" alt="Screen Shot 2021-07-14 at 9 13 52 am" src="https://user-images.githubusercontent.com/6458278/125541876-8f45ae45-a4bc-43e3-9f77-aefbd1a1ecdf.png">

**2. The comment loading spinner border flows over the comment container**

![comment-overflow](https://user-images.githubusercontent.com/6458278/125541883-a75e8c7b-910c-490a-9903-0a145f2944f6.gif)

**3. The comment loading spinner isn't visible in the light theme**

This PR addresses these issues.

ℹ️  **The changes in this PR have been migrated to D63016-code**

#### Does this pull request change what data or activity we track or use?
Nup.

#### Testing instructions:
1. Check that the comment button displays correctly in both dark and light carousel views
2. Submit a comment and ensure the spinner border doesn't overflow. Check in both light and dark themes that the spinner displays correctly.
3.  Add a newspack post carousel to the same page as the gallery and make sure that the next and previous buttons look correct on both ltr and rtl languages.


![light-theme-comment-spinner](https://user-images.githubusercontent.com/6458278/125542455-e3fc8456-35e1-4798-bfa8-67acc56e582b.gif)

ramonjd committed:
[not verified] Define a default color for the dark carousel view
Ensure the comments loading spinner doesn't overflow when submitting a comment

ramonjd committed:
[not verified] added changelog and light theme spinner color change

andrewserong commented on projects/plugins/jetpack/modules/carousel/jetpack-carousel.css:
```diff
@@ -794,6 +795,7 @@ textarea#jp-carousel-comment-form-comment-field:focus::-webkit-input-placeholder
 	margin-top: 20px;
 	color: #999;
 	position: relative;
+	overflow: hidden;
```
Tiny nit: `overflow` is already set a few lines up. Shall we replace that one with `overflow: hidden`?

ramonjd replied:
That would make too much sense! :trollface: 

Will do. Good spotting.

andrewserong reviewed (commented):
This is testing nicely @ramonjd, and good one finding the light mode styling for the spinner 👍

Just left a tiny nit about the overflow attribute. I was wondering too if it's worth updating the `top` position for `#jp-carousel-comment-form-spinner` from `40%` to something like `calc( 50% - 10px )` (or maybe 15px) to get it centred vertically within the container? I noticed it was slightly off centre vertically, but not at all a big deal if you think it's better to leave it as is.

ramonjd committed:
[not verified] Remove already declared overflow

ramonjd commented:
> Just left a tiny nit about the overflow attribute. I was wondering too if it's worth updating the top position for #jp-carousel-comment-form-spinner from 40% to something like calc( 50% - 10px ) (or maybe 15px) to get it centred vertically within the container? I noticed it was slightly off centre vertically, but not at all a big deal if you think it's better to leave it as is.

Very sober assessment. 

I approve! Will add. 

Thank you for testing 🙇

ramonjd committed:
[not verified] recentering comment loading spinner

andrewserong reviewed (dismissed):
Thanks for fixing that up @ramonjd, this LGTM now! Tested on iOS Safari, and on desktop using Safari, FF, Chrome, and Edge, and tested in light and dark mode 👍

ramonjd commented:
Changes incorporated into D63016-code

Commit:
[not verified] # This is a combination of 2 commits.

Add some next/prev button image overrides to stop clash with newspack post carousel block

Specify the opposite left/right position for the arrows in order to overwrite swiperjs styles

ramonjd committed:
[not verified] Specify the opposite left/right position for the arrows in order to overwrite swiperjs styles

andrewserong committed:
Move Jetpack CSS to be enqueued after Swiper CSS to ensure that overrides take precedence

apeatling reviewed (approved):
Looking good :+1:

sdixon194 commented:
Cherry picked to release branch in d1d1b1450d554cee0106398b5c83597683010c9a
